src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -61,33 +58,40 @@ public HttpInListener(AspNetCoreTraceInstrumentationOptions options)
         this.options = options;
     }
 
-    public override void OnEventWritten(string name, object payload)
+    public override void OnEventWritten(string name, object? payload)
     {
+        var activity = Activity.Current;
+
+        if (activity == null)
```

Comment:
Do we need this check? Could we `Activity.Current!`? IIRC, this is already checked at caller side.

Reply:
Fixed: 1aaea5295b0b1d1c41c588bb94fe1e1989845d65